hermeto/core/resolver.py

```diff
@@ -62,17 +62,50 @@ def resolve_packages(request: Request) -> RequestOutput:
 
 
 def _resolve_packages(request: Request) -> RequestOutput:
-    """Run all requested package managers, return their combined output."""
+    """
+    Run all requested package managers, return their combined output.
+
+    Supports individual experimental flags `x-<name>` alongside `dev-package-managers`.
+    """
     _supported_package_managers = _package_managers
-    requested_types = set(pkg.type for pkg in request.packages)
-    if "dev-package-managers" in request.flags:
-        _supported_package_managers = _package_managers | _dev_package_managers
+    requested_types: set[PackageManagerType] = {pkg.type for pkg in request.packages}
+
+    # handle individual experimental package managers toggles
+    experimental_package_managers: set[PackageManagerType] = {
+        type for type in requested_types if type.startswith("x-")
+    }
+    if experimental_package_managers and "dev-package-managers" in request.flags:
+        raise UnsupportedFeature(
+            "Experimental package manager not supported with --dev-package-managers tag.",
+            solution="Switch to using only x-<pkg> please.",
+        )
+    elif experimental_package_managers:
+        requested_types = {
+            cast(PackageManagerType, t.removeprefix("x-")) if t.startswith("x-") else t
+            for t in requested_types
+        }
+        # Normalize each experimental pkg and enable only those dev managers
+        normalized: set[PackageManagerType] = {
```

Comment:
I am sorry, one likely cannot normalize a _package manager_ (they can in theory, I am not sure whether I would be thrilled or terrified by a corresponding type definition). What you normalize here are string literals, I suggest at the very least adding this information to the name (something like `normalized_pm_names`). Now to think of it this stems from the fact that you directly extend a literal type to accommodate for the new names. Maybe a better approach would be to introduce a union type for PackageManagers and ExperimentalPackageManagers?